Include short example of limiting build directories (#13679)
github/docs: content/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-the-codeql-workflow-for-compiled-languages.md

If you added manual build steps for compiled languages and {% data variables.product.prodname_code_scanning %} is still not working on your repository, contact {% data variables.contact.contact_support %}.

## Excluding directories from the database by customizing the build steps

Sometimes you do not want to build and analyze your whole project and its dependencies. You can omit build dependencies from the {% data variables.product.prodname_codeql %} database by separating their build from the main build that's tracked by {% data variables.product.prodname_codeql %}.

### Example

The following example illustrates this approach for a C++ project that's built using `make`, but the procedure applies to all compiled languages and build systems.

Your project tree contains three subdirectories, `src/`, `lib1`, and `lib2` and your default build command, `make all`, builds them all. If you configure the workflow with the default build command, it generates a {% data variables.product.prodname_codeql %} database with the code and dependencies of all three subdirectories.

> ```sh
> codeql database create cpp-database --language=cpp --command='make all'
> ```
> 
> You can omit one of the subdirectories and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before calling the {% data variables.product.prodname_codeql %} CLI.
> 
> ```sh
> make lib2 
> 
> codeql database create cpp-database --language=cpp --command='make src lib1'
> ```

Reviewer: Would this be an appropriate replacement for someone using the CodeQL action instead of the CLI?

````suggestion
```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    - run: |
       make all

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

You can omit a subdirectory and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before the "Initialize {% data variables.product.prodname_codeql %}" step.

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Build lib2
     - run: |
       make lib2

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    - run: |
       make src lib1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```
````

Author: The action makes this more brittle.  In the first configuration, remove the 
```
    - run: |
       make all
```
part, that will be run by codeql.

In the second configuration, remove 
```
    - run: |
       make src lib1
```
because that will also be run by the codeql runner.

Reviewer: Thank you. Would I need to put the autobuild part back instead?

That is:
```
    - name: Initialize CodeQL
	    uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}
	
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1
	
    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

Reviewer: Thanks.

````suggestion
```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    # Autobuild attempts to build any compiled languages
    # In this case it runs the default build command `make all`
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

You can omit a subdirectory and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before the "Initialize {% data variables.product.prodname_codeql %}" step.

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Build lib2
     - run: |
       make lib2

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    # Autobuild attempts to build any compiled languages
    # In this case it skips making `lib2` as this is already built
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```
````

For more tips and tricks about why `autobuild` won't build your code, see "[Troubleshooting the {% data variables.product.prodname_codeql %} workflow](/code-security/secure-coding/troubleshooting-the-codeql-workflow)."

If you added manual build steps for compiled languages and {% data variables.product.prodname_code_scanning %} is still not working on your repository, contact {% data variables.contact.contact_support %}.

## Excluding directories from the database by customizing the build steps

Sometimes you do not want to build and analyze your whole project and its dependencies. You can omit build dependencies from the {% data variables.product.prodname_codeql %} database by separating their build from the main build that's tracked by {% data variables.product.prodname_codeql %}.

### Example

The following example illustrates this approach for a C++ project that's built using `make`, but the procedure applies to all compiled languages and build systems.

> Your project tree contains three subdirectories, `src/`, `lib1`, and `lib2` and your default build command, `make all`, builds them all. If you configure the workflow with the default build command, it generates a {% data variables.product.prodname_codeql %} database with the code and dependencies of all three subdirectories.

Reviewer: ```suggestion
Your project tree contains three subdirectories, `src/`, `lib1`, and `lib2` and your default build command, `make all`, builds them all. In the following workflow, the "Autobuild" step will run the default build command and generate a {% data variables.product.prodname_codeql %} database with the code and dependencies of all three subdirectories.
```

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      
    - name: Initialize CodeQL
      uses: github/codeql-action/init@v1
      with:
        languages: ${{ matrix.language }}

    # Autobuild attempts to build any compiled languages
    # In this case it runs the default build command `make all`
    - name: Autobuild
      uses: github/codeql-action/autobuild@v1

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

You can omit a subdirectory and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before the "Initialize {% data variables.product.prodname_codeql %}" step.

```yml
    steps:
    - name: Checkout repository
      uses: actions/checkout@v2
      

>     - name: Build lib2

Reviewer: Add a comment
```suggestion
    # Build `lib2` before initializing CodeQL
    - name: Build lib2
```

### Example

The following example illustrates this approach for a C++ project that's built using `make`, but the procedure applies to all compiled languages and build systems.

Your project tree contains three subdirectories, `src/`, `lib1`, and `lib2` and your default build command, `make all`, builds them all. If you configure the workflow with the default build command, it generates a {% data variables.product.prodname_codeql %} database with the code and dependencies of all three subdirectories.

```sh
codeql database create cpp-database --language=cpp --command='make all'
```

You can omit one of the subdirectories and its dependencies from the {% data variables.product.prodname_codeql %} database by building it before calling the {% data variables.product.prodname_codeql %} CLI.

```sh
make lib2 

codeql database create cpp-database --language=cpp --command='make src lib1'
```

The resulting {% data variables.product.prodname_codeql %} database will contain only code and dependencies from the `src` and `lib1` directories.

{% note %}

> **Note:** This approach does not work for header-only C++ libraries.

Reviewer: @hohn - is there an alternative approach that would work for header-only C++ libraries? If not, can/should we say why this doesn't work for them?

Reviewer 2: it is base KLAY coin  main net

Reviewer 2: I went changing  wallat address and need contol table.
the first is my wallet address
0x4C927A35D98b37af8f6Ce853165573Ab13a6383A    .
Whoever controls it 1% Owner 1% Each Developer 1% Load Architect 1% Liquidity Pool 60% Sales Transaction 30% Rest Burn and Server Maintenance Improvement Cost.

Reviewer 2: We make test erc20 코인 메타마스크 워렛에서 만드는것 공시좀

전체의 공유만드는 지갑주소를 만들어주세요

Reviewer 2: 이더기반 메타마스크 이용 하여만들어요.
그리고 전체에게 지갑생성을위한 왈렛주소 주기바랍니다.
